libraries/Microsoft.Bot.StreamingExtensions/HttpContentStream.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net.Http;
+
+namespace Microsoft.Bot.StreamingExtensions
+{
+    /// <summary>
+    /// An attachment contained within a <see cref="StreamingRequest"/>'s stream collection,
+    /// which itself contains any form of media item.
+    /// </summary>
+    public class HttpContentStream
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HttpContentStream"/> class.
+        /// and assigns an unique guid as its Id.
+        /// </summary>
+        public HttpContentStream()
+            : this(Guid.NewGuid())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HttpContentStream"/> class.
+        /// </summary>
+        /// <param name="id">A <see cref="Guid"/> to assign as the Id of this instance of <see cref="HttpContentStream"/>.
+        /// If null a new <see cref="Guid"/> will be generated.
+        /// </param>
+        public HttpContentStream(Guid id)
+        {
+            if (id != null)
+            {
+                Id = id;
+            }
+            else
+            {
+                Id = Guid.NewGuid();
+            }
+        }
+
+        /// <summary>
+        /// Gets the <see cref="Guid"/> used to identify this <see cref="HttpContentStream"/>.
+        /// </summary>
+        /// <value>
+        /// A <see cref="Guid"/> used to identify this <see cref="HttpContentStream"/>.
+        /// </value>
+        public Guid Id { get; private set; }
+
```

Comment:
Why do we need globally unique http Content Stream Ids? What's wrong with "1" or "Stream1" or something simple?

Reply:
Stream IDs end up being the identifier for the assemblers building them as they come in, the less unique they are the more compound the assembler IDs will become. Also they have to fit into the same ID space in the Header as the other message types.